drivers/i2c/i2c_rv32m1_lpi2c.c does not compile

I noted that during CI process the rv32m1_vega_zero_riscy does not get compiled:

```
/home/visconti/Projects/GIT/FW/zephyrproject/zephyr/drivers/i2c/i2c_rv32m1_lpi2c.c
In file included from ../include/zephyr/sys/util_macro.h:34,
                 from ../include/zephyr/irq_multilevel.h:15,
                 from ../include/zephyr/devicetree.h:20,
                 from ../include/zephyr/device.h:12,
                 from ../include/zephyr/drivers/i2c.h:25,
                 from /local/home/visconti/Projects/GIT/FW/zephyrproject/zephyr/drivers/i2c/i2c_rv32m1_lpi2c.c:13:
/local/home/visconti/Projects/GIT/FW/zephyrproject/zephyr/drivers/i2c/i2c_rv32m1_lpi2c.c:271:34: warning: implicit declaration of function 'INST_DT_CLOCK_IP_NAME' [-Wimplicit-function-declaration]
  271 |                 .clock_ip_name = INST_DT_CLOCK_IP_NAME(id),                    \
      |                                  ^~~~~~~~~~~~~~~~~~~~~

```

You can reproduce it with some shields, or, very easily, adding ```CONFIG_I2C=y``` to hello_world prj.conf and then:

```
west build -b rv32m1_vega_zero_riscy samples/hello_world/
```

--- a/drivers/i2c/i2c_rv32m1_lpi2c.c
+++ b/drivers/i2c/i2c_rv32m1_lpi2c.c
@@ -15,6 +15,7 @@
 #include <zephyr/irq.h>
 #include <fsl_lpi2c.h>
 #include <zephyr/logging/log.h>
+#include <soc.h>
 #include <zephyr/drivers/pinctrl.h>
 
 LOG_MODULE_REGISTER(rv32m1_lpi2c);

